src/Components/Web/src/Web/ErrorBoundary.cs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+
+namespace Microsoft.AspNetCore.Components.Web
+{
+    /// <summary>
+    /// Captures errors thrown from its child content.
+    /// </summary>
+    public class ErrorBoundary : ErrorBoundaryBase
+    {
+        [Inject] private IErrorBoundaryLogger? ErrorBoundaryLogger { get; set; }
+
+        /// <summary>
+        /// Invoked by the base class when an error is being handled. The default implementation
+        /// logs the error.
+        /// </summary>
+        /// <param name="exception">The <see cref="Exception"/> being handled.</param>
+        protected override async Task OnErrorAsync(Exception exception)
+        {
+            await ErrorBoundaryLogger!.LogErrorAsync(exception);
+        }
```

Comment:
I think we should reconsider this and make it synchronous. Making it asynchronous introduces an unnecessary level of complexity and I can't think of scenarios in which this is useful, we do fire and forget for logging in other places and it should be doable here.

Reply:
Seems like a pros and cons thing. I did start with synchronous only, but switched to allow async because:

 * The total amount of implementation complexity this adds for us is pretty modest (it's [these lines](https://github.com/dotnet/aspnetcore/pull/30874/files#diff-7b31bfe2274cde39c02bc253d4b715c96ce0d5448a749f35b5bcdb1dc3eb05d5R112-R122))
 * If we can make it easier for a developer to do async logging right, why wouldn't we? If we don't do it, we're just pushing the responsibility across to developers, who might not have a good sense of how to cause async errors to show up anywhere.
 * If people don't need async, it's very easy for them to do sync-only operations within this. Whereas the inverse is not true. And if we eventually had to have both `OnError` and `OnErrorAsync`, that's certainly more conceptual weight.

If you feel really strongly please say so, otherwise I'd err on the side of being flexible for developers here.

Reply:
> * And if we eventually had to have both `OnError` and `OnErrorAsync`, that's certainly more conceptual weight.

That's the main reason I can think of, however for me the drawback is having to think about async errors that might happen within the errorboundary `OnErrorAsync`

I don't feel strongly about it, I just thought it was worth a conversation.